azurerm/resource_arm_netapp_pool.go

```diff
@@ -0,0 +1,220 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/netapp/mgmt/2019-06-01/netapp"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmNetAppPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmNetAppPoolCreateUpdate,
+		Read:   resourceArmNetAppPoolRead,
+		Update: resourceArmNetAppPoolCreateUpdate,
+		Delete: resourceArmNetAppPoolDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[\da-zA-Z_-]{4,64}$`),
+					`The name must be between 4 and 64 characters in length and contain only letters, numbers, underscore or hyphens.`,
+				),
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`(^[\da-zA-Z])([-\da-zA-Z]{1,62})([\da-zA-Z]$)`),
+					`The name must be between 3 and 64 characters in length and begin with a letter or number, end with a letter or number and may contain only letters, numbers.`,
+				),
+			},
+
+			"service_level": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(netapp.Premium),
+					string(netapp.Standard),
+					string(netapp.Ultra),
+				}, false),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"size": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(4398046511104, 549755813888000),
```

Comment:
I'd suggest we make this a little easier on users and have this be `size_in_4_tb` and then 
```suggestion
				ValidateFunc: validation.IntBetween(1, 30),
```
at that point we can multiply any number passed in by `4398046511104` when we read it and then divide by `4398046511104` when we set it.

Reply:
fixed